### ISSUE TITLE
Error recommends re-linking?

### ISSUE THREAD

**Issue by jakimfett** [ohwgiles/laminar#99]
Any idea what would cause this?

> laminar@remoteserver:~/app/cfg/jobs$ laminard --help
> laminard: Symbol `_ZTVN5capnp18TwoPartyVatNetworkE' has different size in shared object, consider re-linking
terminate called after throwing an instance of 'kj::ExceptionImpl'
>  what():  kj/filesystem.c++:612: failed: directory does not exist; path = var/lib/laminar
> Aborted

Aside from this, seems to be working, laminar is online and I can view builds/jobs/graphs/etc.

_edit: a name_

**Comment by ohwgiles** [comment]
> laminard: Symbol `_ZTVN5capnp18TwoPartyVatNetworkE' has different size in shared object, consider re-linking

This looks like `laminard` was built against a different version of capnproto than the version it's running against. To check that, compare the library names in `ldd laminard` with those in `CMakeCache.txt` from laminar's build directory

> what(): kj/filesystem.c++:612: failed: directory does not exist; path = var/lib/laminar

You get this error and laminar is still working? Are you sure?

**Comment by jakimfett** [comment]
> @ohwgiles commented:
> ...built against a different version of capnproto...

That's my problem, thanks.
I had rebuilt the capnproto job, but laminar hadn't been rebuilt since then.

**Tangentially:**  
Is there currently a way to set up jobs to cascade dependent builds on success?

> ohwgiles also commented:
> ...and laminar is still working? Are you sure?

Technically no, but functionally yes.
Laminar runs exactly until this happens (which could be triggered by attempting to use auto-completion, as well as executing the binary itself). 

My testbed system is set to aggressively restart services when they fail, and until I looked more closely, I did not realize that the service was dying and then immediately being resurrected my reaper.

---
Thanks for the feedback, I'm closing this because imo it's a solved problem for me at this point.
